modules/images/regenerate-existing-images/background-process/class-perflab-background-process.php

```diff
@@ -0,0 +1,236 @@
+		 * real cron has been setup to do so.
+		 */
+		if ( defined( 'ENABLE_BG_PROCESS_CRON' ) ) {
+			return;
+		}
+
+		$nonce  = wp_create_nonce( self::BG_PROCESS_ACTION );
+		$job_id = $this->job->job_id;
+
+		$url    = admin_url( 'admin-ajax.php' );
+		$params = array(
+			'blocking'  => false,
+			'body'      => array(
+				'action' => self::BG_PROCESS_ACTION,
+				'job_id' => $job_id,
+				'nonce'  => $nonce,
+			),
+			'sslverify' => apply_filters( 'https_local_ssl_verify', false ),
+			'timeout'   => 0.1,
+		);
+
+		wp_remote_post( $url, $params );
+	}
+
+	/**
+	 * Checks whether the memory is exceeded for the current process.
+	 *
+	 * Keep memory limit to 90% of the values assigned to PHP config,
+	 * it will prevent the error and help run the memory exhaustion check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	private function memory_exceeded() {
+		$memory_limit     = $this->get_memory_limit() * 0.9; // Use 90% of memory limit.
+		$memmory_usage    = memory_get_usage( true ); // Memory allotted to php.
+		$memory_exhausted = false;
+
+		// Check if memory usage has reached the memory limit (90%).
+		if ( $memmory_usage >= $memory_limit ) {
+			$memory_exhausted = true;
+		}
+
+		/**
+		 * Filters the memory exceeded flag.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param bool $memory_exhausted Whether the memory limit has been reached.
+		 */
+		return apply_filters( 'perflab_background_process_memory_exceeded', $memory_exhausted );
+	}
+
+	/**
+	 * Get the memory limit allotted for PHP.
+	 *
+	 * Keep the default memory limit of 128M which
+	 * is there on most of the hosts.
+	 *
+	 * @return int
+	 */
+	private function get_memory_limit() {
+		$default_memory_limit = '128M';
+
+		if ( function_exists( 'memory_limit' ) ) {
+			$memory_limit = ini_get( 'memory_limit' );
+		}
+
+		if ( empty( $memory_limit ) || -1 === $memory_limit ) {
+			$memory_limit = $default_memory_limit;
+		}
+
+		// Convert memory to bytes in integer format.
+		return wp_convert_hr_to_bytes( $memory_limit );
+	}
+
+	/**
+	 * Checks if current execution time is exceeded for the process.
+	 *
+	 * A sensible default of 20 seconds has been taken in case ini_get does not
+	 * work. Keep max_execution_time to 10 seconds less than whatever is set in PHP
+	 * configuration, so that those 10 seconds can be used to cleanup everything and
+	 * call the next batch of job.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	private function time_exceeded() {
+		$current_time       = time();
+		$run_start_time     = $this->job->get_start_time();
+		$min_execution_time = 20; // Default to 20 seconds. Almost, all servers will have this much of time.
+
+		if ( function_exists( 'ini_get' ) ) {
+			$time               = ini_get( 'max_execution_time' );
+			$max_execution_time = ( ! empty( $time ) && ( $time > $min_execution_time ) ) ? $time - 10 : $min_execution_time;
+		}
+
+		return ( $current_time >= ( $run_start_time + $max_execution_time ) );
```

Comment:
@ankitrox we should also use `apply_filters` on the value returned here as we have with the memory limit. The filter should be `perflab_background_process_time_exceeded`
```suggestion
		$time_exceeded = ( $current_time >= ( $run_start_time + $max_execution_time ) );
		return apply_filters( 'perflab_background_process_time_exceeded', $time_exceeded );
```